### ISSUE TITLE
Add a Mix and Render option which cannot clip or a master gain

### ISSUE THREAD

**Issue by LWinterberg** [audacity/audacity#1791]
(this is from the todo.txt) 

Mix and render currently simply plonks the tracks onto each other with their respectively set volumes, regardless of whether they clip or not. Having a thing which adjusts the gain automatically to the point where things no longer clip may be wanted (or a master gain so you can manually set it down a notch or two after you notice clipping)

**Comment by petersampsonaudacity** [comment]
See this long-standing proposal from the Audacity Wiki:

https://wiki.audacityteam.org/wiki/Proposal_Easy_Avoid_Clipping_on_many_tracks
Proposal Easy Avoid Clipping on many tracks

_Ir remains one of the most popular Feature Requests in the Wiki - with currently 30 votes (which is a lot as the Wiki goes)_

**Comment by SteveDaulton** [comment]
"Mix tracks without clipping" is easy to achieve with a macro.
It would be nice if macro commands could be made into menu commands 
(Create a macro, then add the macro to the menus as "Tracks menu > Mix > Mix and Normalize").

A "master volume" slider in the MixerBoard would imo be more useful / versatile, or a way to create "submixes" (as found on larger hardware mixers).

**Comment by werame** [comment]
@SteveDaulton If it can be done with a macro, can't it be done with Nyquist one? Those can be added to the menu. Or would this macro need to call Nyquist effects, which is blocked by #2465 ?

**Comment by SteveDaulton** [comment]
> @SteveDaulton If it can be done with a macro, can't it be done with Nyquist one (`;type tool`)?

Yes it can be done with Nyquist, but Macros are easier for most users.

This works as a Macro:
```
SelectAll:
MixAndRender:
Normalize:ApplyGain="1" PeakLevel="-1" RemoveDcOffset="1" StereoIndependent="0"
```

This works in Nyquist:
```
;type tool
(aud-do "SelectAll:")
(aud-do "MixAndRender:")
(aud-do "Normalize:ApplyGain=1 PeakLevel=-1 RemoveDcOffset=1 StereoIndependent=0")
"" ;Return no-op
```
Both require that the default sample format is 32-bit float (otherwise there's likely to be clipping damage).

Even if we shipped a "Mix and Normalize" Macro or Plug-in, I still think that a master volume slider in the "Mixer Board" would be very useful.

**Comment by LWinterberg** [comment]
duplicate of #1732
